### PR TITLE
feat(css/minifier): merge rules

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Merge the same rules + all tests from https://github.com/cssnano/cssnano/tree/master/packages/postcss-merge-rules

**BREAKING CHANGE:**

No

**Related issue (if exists):**

partial https://github.com/swc-project/swc/issues/5880, there are two todos to solve all cases:
- future browserslist support
- partial merging rules (will be iin near future)

### CODE REVIEW

File: crates/swc_css_minifier/src/compressor/rules.rs [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+        }
+
+        // Merge when both selectors are exactly equal
+        // e.g. a { color: blue } a { font-weight: bold }
+        if left.prelude.eq_ignore_span(&right.prelude) {
+            let block = self.merge_simple_block(&left.block, &right.block);
+            let mut qualified_rule = QualifiedRule {
+                span: Span::new(
+                    left.span.span_lo(),
+                    right.span.span_lo(),
```
Comment (kdy1) [inline crates/swc_css_minifier/src/compressor/rules.rs:230]: I think this is a mistake?

Comment (alexander-akait) [reply crates/swc_css_minifier/src/compressor/rules.rs:230]: Yeah, typo, need fix

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - swc_css_minifier

File: crates/swc_css_minifier/src/compressor/rules.rs (hunk 1 above)
Comment (kdy1) [reply crates/swc_css_minifier/src/compressor/rules.rs]: I fixed it

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI failed

**Review by kdy1 — CHANGES_REQUESTED** [review]
Can you rebase again to exclude unrelated commits?

**Comment by alexander-akait** [comment]
@kdy1 Yeah, WIP on this, my mistake

**Comment by alexander-akait** [comment]
@kdy1 Fixed

**Review by kdy1 — APPROVED** [review]

**Comment by ludofischer** [comment]
Does the method in this PR also solve the various issues with the merge-rules implementation in cssnano (https://github.com/cssnano/cssnano/projects/5)?

**Comment by alexander-akait** [comment]
@ludofischer we don't implement the partial merging and browserslist based, so we don't have these problems right now (but yes, we will need to solve them in future)

**Pull request merged**